hazelcast/include/hazelcast/client/sql/sql_page.h

```diff
@@ -39,15 +39,47 @@ namespace sql {
 /**
  * A finite set of rows returned to the client.
  */
-class HAZELCAST_API sql_page : public std::enable_shared_from_this<sql_page>
+class HAZELCAST_API sql_page
 {
     using column = std::vector<boost::any>;
+    struct HAZELCAST_API shared_block
+    {
```

Comment:
It is unfortunately not possible because definition of `page_data` must be known by `sql_row` that's why I defined it there.

I can move it to `pimpl` namespace at the outer of the class, if it is required.

Reply:
Fixed.